src/syscalls/alt_bn128/compression.rs

```diff
@@ -0,0 +1,601 @@
+        }
+    }
+
+    pub fn alt_bn128_g2_decompress(
+        input: &[u8; G2_COMPRESSED],
+    ) -> Result<[u8; G2], AltBn128CompressionError> {
+        let mut result_buffer = [0; G2];
+        let result = unsafe {
+            crate::syscalls::sol_alt_bn128_compression(
+                ALT_BN128_G2_DECOMPRESS,
+                input as *const _ as *const u8,
+                input.len() as u64,
+                &mut result_buffer as *mut _ as *mut u8,
+            )
+        };
+
+        match result {
+            0 => Ok(result_buffer),
+            error => Err(AltBn128CompressionError::from(error)),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+
+    use super::target_arch::convert_endianness;
+
+    use {
+        super::*,
+        // crate::convert_endianness,
+        ark_serialize::{CanonicalDeserialize, CanonicalSerialize, Compress, Validate},
+        std::ops::Neg,
+        target_arch::{
+            alt_bn128_g1_compress, alt_bn128_g1_decompress, alt_bn128_g2_compress,
+            alt_bn128_g2_decompress,
+        },
+    };
+    type G1 = ark_bn254::g1::G1Affine;
+    type G2 = ark_bn254::g2::G2Affine;
+    use ark_ff::UniformRand;
+    use ark_std::test_rng;
+
+    #[test]
+    fn alt_bn128_g1_compression() {
+        let g1_be = [
+            45, 206, 255, 166, 152, 55, 128, 138, 79, 217, 145, 164, 25, 74, 120, 234, 234, 217,
+            68, 149, 162, 44, 133, 120, 184, 205, 12, 44, 175, 98, 168, 172, 20, 24, 216, 15, 209,
+            175, 106, 75, 147, 236, 90, 101, 123, 219, 245, 151, 209, 202, 218, 104, 148, 8, 32,
+            254, 243, 191, 218, 122, 42, 81, 193, 84,
+        ];
+
+        let g1_le = convert_endianness::<32, 64>(&g1_be);
+        let g1: G1 =
+            G1::deserialize_with_mode(g1_le.as_slice(), Compress::No, Validate::No).unwrap();
+
+        let g1_neg = g1.neg();
+        let mut g1_neg_be = [0u8; 64];
+        g1_neg
+            .x
+            .serialize_with_mode(&mut g1_neg_be[..32], Compress::No)
+            .unwrap();
+        g1_neg
+            .y
+            .serialize_with_mode(&mut g1_neg_be[32..64], Compress::No)
+            .unwrap();
+        let g1_neg_be: [u8; 64] = convert_endianness::<32, 64>(&g1_neg_be);
+
+        let points = [(g1, g1_be), (g1_neg, g1_neg_be)];
+
+        for (point, g1_be) in &points {
+            let mut compressed_ref = [0u8; 32];
+            G1::serialize_with_mode(point, compressed_ref.as_mut_slice(), Compress::Yes).unwrap();
+            let compressed_ref: [u8; 32] = convert_endianness::<32, 32>(&compressed_ref);
+            println!("point {:?}", point.to_flags());
+            // assert_eq!(compressed_ref, expected);
+
+            let decompressed = alt_bn128_g1_decompress(compressed_ref.as_slice()).unwrap();
+
+            assert_eq!(
+                alt_bn128_g1_compress(&decompressed).unwrap(),
+                compressed_ref
+            );
+            assert_eq!(decompressed, *g1_be);
+        }
+    }
+
+    #[ignore]
```

Comment:
Any reason why do we ignore this test? If there is some future work remaining, I would leave a TODO comment.

Reply:
It's too heavy to execute with this many iterations I will reduce the number of iterations.

Reply:
reduced to 1k and 100 iterations so that execution time is bearable.